src/System Application/App/SharePoint Authorization/src/SharePointAuth.Codeunit.al

```diff
@@ -82,4 +82,37 @@ codeunit 9142 "SharePoint Auth."
     begin
         exit(SharePointAuthImpl.CreateAuthorizationCode(EntraTenantId, ClientId, ClientSecret, Scopes));
     end;
+
+    /// <summary>
+    /// Creates an authorization mechanism with the Client Credentials Grant Flow.
+    /// </summary>
+    /// <param name="AadTenantId">Azure Active Directory tenant ID</param>
+    /// <param name="ClientId">The Application (client) ID that the Azure portal - App registrations experience assigned to your app.</param>        
+    /// <param name="Certificate">The Base64-encoded certificate for the Application (client) configured in the Azure Portal - Certificates &amp; Secrets.</param>
+    /// <param name="CertificatePassword">Password for the certificate.</param>
+    /// <param name="Scope">A scope that you want the user to consent to.</param>
+    /// <returns>Codeunit instance implementing authorization interface</returns>
+    procedure CreateClientCredentials(AadTenantId: Text; ClientId: Text; Certificate: Text; CertificatePassword: SecretText; Scope: Text): Interface "SharePoint Authorization";
```

Comment:
use Secret text for the certificate too

Reply:
The procedures in the OAuth2 Codeunit doesn't use SecretText for the Certificate either.
@darjoo how should I proceed with this?

Reply:
I'll update the OAuth2 module such that it takes in SecretText for the Certificate. Shouldn't take too long, I'll update your PR once I have that in as well.

Reply:
I think this one addresses this issue: #792 🙂